### PR TITLE
[20492] OpenOutputChannels / CloseOutputChannels that receive a LocatorSelectorEntry

### PR DESCRIPTION
**Pull request by jepemi**
## Description

Introduce new `OpenOutputChannels` and `CloseOutputChannels` methods to the `TransportInterface` base class, both accepting a `LocatorSelectorEntry` argument. This enhancement provides improved control over creating or destroying communication channels between participants.

- This PR depends on  #4300 and must be merged after that one. 

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- ❌ Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: include/fastdds/rtps/transport/TransportInterface.h [hunk 1]
```diff
@@ -135,6 +135,18 @@ class RTPS_DllAPI TransportInterface
             SendResourceList& sender_resource_list,
             const Locator&) = 0;
 
+    //! Must open the channel that maps to/from the given locator selector entry. This method must allocate,
+    //! reserve and mark any resources that are needed for said channel.
+    virtual bool OpenOutputChannel(
+            SendResourceList& sender_resource_list,
+            const fastrtps::rtps::LocatorSelectorEntry&);
```
Comment (MiguelCompany) [inline include/fastdds/rtps/transport/TransportInterface.h:142]: Add `#include <fastdds/rtps/common/LocatorSelectorEntry.hpp>`

File: src/cpp/rtps/transport/TransportInterface.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fastdds/rtps/transport/TransportInterface.h>
+
+#include <fastdds/rtps/common/LocatorSelector.hpp>
```
Comment (MiguelCompany) [inline src/cpp/rtps/transport/TransportInterface.cpp:17]:
```suggestion
#include <fastdds/rtps/common/LocatorSelectorEntry.hpp>
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastdds/rtps/transport/TransportInterface.h [hunk 3]
```diff
@@ -135,6 +136,31 @@ class RTPS_DllAPI TransportInterface
...
+            SendResourceList& sender_resource_list,
+            const fastrtps::rtps::LocatorSelectorEntry& locator_selector_entry);
+
+    /**
+     * Close the channel that maps to/from the given locator selector entry.
+     *
+     * @param sender_resource_list Participant's send resource list.
+     * @param locator_selector_entry Locator selector entry with the remote participant's locators.
+     *
+     * @return true if the channel was correctly closed or if it was already closed.
```
Comment (MiguelCompany) [inline include/fastdds/rtps/transport/TransportInterface.h:158]:
```suggestion
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — DISMISSED** [review]

**Review by MiguelCompany — DISMISSED** [review]
LGTM with green CI

File: include/fastdds/rtps/transport/TransportInterface.h [hunk 4]
```diff
@@ -135,6 +136,29 @@ class RTPS_DllAPI TransportInterface
             SendResourceList& sender_resource_list,
             const Locator&) = 0;
 
+    /**
+     * Must open the channel that maps to/from the given locator selector entry. This method must allocate,
+     * reserve and mark any resources that are needed for said channel.
+     *
+     * @param sender_resource_list Participant's send resource list.
+     * @param locator_selector_entry Locator selector entry with the remote participant's locators.
```
Comment (MiguelCompany) [inline include/fastdds/rtps/transport/TransportInterface.h:144]:
```suggestion
     * @param locator_selector_entry Locator selector entry with the remote entity locators.
```

File: include/fastdds/rtps/transport/TransportInterface.h [hunk 5]
```diff
@@ -135,6 +136,29 @@ class RTPS_DllAPI TransportInterface
...
+     */
+    virtual bool OpenOutputChannel(
+            SendResourceList& sender_resource_list,
+            const fastrtps::rtps::LocatorSelectorEntry& locator_selector_entry);
+
+    /**
+     * Close the channel that maps to/from the given locator selector entry.
+     *
+     * @param sender_resource_list Participant's send resource list.
+     * @param locator_selector_entry Locator selector entry with the remote participant's locators.
```
Comment (MiguelCompany) [inline include/fastdds/rtps/transport/TransportInterface.h:156]:
```suggestion
     * @param locator_selector_entry Locator selector entry with the remote entity locators.
```

File: src/cpp/rtps/transport/TransportInterface.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fastdds/rtps/transport/TransportInterface.h>
+
+#include <fastdds/rtps/common/LocatorSelectorEntry.hpp>
+
+
```
Comment (MiguelCompany) [inline src/cpp/rtps/transport/TransportInterface.cpp:19]:
```suggestion

```

**Review by MiguelCompany — APPROVED** [review]

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Accidentally approved

**Review by MiguelCompany — APPROVED** [review]

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by EduPonz** [comment]
@richiprosima please test this

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Comment by EduPonz** [comment]
@richiprosima please test this

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**